lib/simplejit/src/backend.rs

```diff
@@ -40,14 +42,49 @@ impl SimpleJITBuilder {
     /// instead.
     pub fn with_isa(isa: Box<TargetIsa>) -> Self {
         debug_assert!(!isa.flags().is_pic(), "SimpleJIT requires non-PIC code");
-        Self { isa }
+        let symbols = HashMap::new();
+        Self { isa, symbols }
+    }
+
+    /// Define a symbol in the internal symbol table.
+    ///
+    /// The JIT will use the symbol table to resolve names that are declared,
+    /// but not defined, in the module being compiled.  A common example is
+    /// external functions.  With this method, functions and data can be exposed
+    /// to the code being compiled which are defined by the host.
+    ///
+    /// If a symbol is defined more than once, the most recent definition will
+    /// be retained.
+    ///
+    /// If the JIT fails to find a symbol in its internal table, it will fall
+    /// back to a platform-specific search (this typically involves searching
+    /// the current process for public symbols, followed by searching the
+    /// platform's C runtime).
+    pub fn symbol<'a>(&'a mut self, name: &str, ptr: *const u8) -> &'a mut Self {
```

Comment:
No need to qualify; I agree.  I changed it to `K: Into<String>` in order to match the signature of `symbols`.  The main reason I used that on `symbols` in the first place was just to make it slightly easier to set up pre-defined tables (*i.e.* you know all the symbols you want to expose, so you've got `&str`s).

That said, if you'd prefer to avoid generics, I can change them both to just plain old `String`.